### PR TITLE
Set IsTabStop to false when tabIndex is negative

### PR DESCRIPTION
**Pull request by lamxdoan**
Fix issue 1 in #2784 

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/microsoft/react-native-windows/pull/4180)

### CODE REVIEW

**Comment by kmelmon** [comment]
>   int32_t m_tabIndex = std::numeric_limits<std::int32_t>::max();

Should this be initialized to -1?

---
Refers to: vnext/ReactUWP/Views/ViewViewManager.cpp:215 in 8fd8adc. [](commit_id = 8fd8adc03836166e48f719b94551df67f72c9a63, deletion_comment = False)

File: vnext/ReactUWP/Views/ControlViewManager.cpp [hunk 1]
```diff
@@ -64,8 +64,14 @@ void ControlViewManager::UpdateProperties(ShadowNodeBase *nodeToUpdate, const fo
       } else if (propertyName == "tabIndex") {
         if (propertyValue.isNumber()) {
           auto tabIndex = propertyValue.asDouble();
-          if (tabIndex == static_cast<int32_t>(tabIndex))
-            control.TabIndex(static_cast<int32_t>(tabIndex));
+          if (tabIndex == static_cast<int32_t>(tabIndex)) {
+            if (tabIndex < 0) {
+              control.IsTabStop(false);
```
Comment (kmelmon) [inline vnext/ReactUWP/Views/ControlViewManager.cpp]: >control.IsTabStop(false); [](start = 14, length = 25)

In addition to setting IsTabStop = false, consider clearing the TabIndex property.

File: vnext/ReactUWP/Views/ViewViewManager.cpp [hunk 2]
```diff
@@ -87,8 +87,14 @@ class ViewShadowNode : public ShadowNodeBase {
   void TabIndex(int32_t tabIndex) {
     m_tabIndex = tabIndex;
 
-    if (IsControl())
-      GetControl().TabIndex(m_tabIndex);
+    if (IsControl()) {
+      if (tabIndex < 0) {
+        GetControl().IsTabStop(false);
```
Comment (kmelmon) [inline vnext/ReactUWP/Views/ViewViewManager.cpp]: >GetControl().IsTabStop(false); [](start = 8, length = 30)

In addition to setting IsTabStop = false, consider clearing the TabIndex property.

**Comment by lamxdoan** [comment]
>   int32_t m_tabIndex = std::numeric_limits<std::int32_t>::max();

Yeah I think this is a more reasonable default than max int

---
In reply to: [592181044](https://github.com/microsoft/react-native-windows/pull/4180#issuecomment-592181044) [](ancestors = 592181044)

---
Refers to: vnext/ReactUWP/Views/ViewViewManager.cpp:215 in 8fd8adc. [](commit_id = 8fd8adc03836166e48f719b94551df67f72c9a63, deletion_comment = False)

File: vnext/ReactUWP/Views/ControlViewManager.cpp (hunk 1 above)
Comment (lamxdoan) [reply vnext/ReactUWP/Views/ControlViewManager.cpp]: This makes sense. Now clearing the property and tested that everything continues to work.

---
In reply to: [385377044](https://github.com/microsoft/react-native-windows/pull/4180#discussion_r385377044) [](ancestors = 385377044)

File: vnext/ReactUWP/Views/ViewViewManager.cpp (hunk 2 above)
Comment (lamxdoan) [reply vnext/ReactUWP/Views/ViewViewManager.cpp]: This makes sense. Now clearing the property and tested that everything continues to work.

---
In reply to: [385377113](https://github.com/microsoft/react-native-windows/pull/4180#discussion_r385377113) [](ancestors = 385377113)

**Pull request merged**